rcl/include/rcl/logging.h

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__LOGGING_H_
+#define RCL__LOGGING_H_
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "rcl/visibility_control.h"
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+/// The flag if the logging system has been initialized.
+RCL_PUBLIC
+extern bool g_rcl_logging_initialized;
+
+/// Initialize the logging system.
+/**
+ * The function is called automatically when using the logging macros.
+ */
+RCL_PUBLIC
+void rcl_logging_initialize();
+
+/// The structure identifying the caller location in the source code.
+typedef struct RclLogLocation
```

Comment:
I started with names similar as in ROS 1. Updated in 42a02ab383b8d8884d93f9e322a4b05ee91bd10b